server/functions.php

```diff
@@ -303,3 +303,84 @@ function get_user_by_ssn($ssn) {
 		);
 	}
 }
+
+if(!function_exists("get_lectures_by_course")){
+	function get_lectures_by_course($courseId, $startTs = null, $endTs = null){
+		$pdo = new PDO("sqlite:../db.sqlite");
+
+		// Get type of user
+		$userData = get_myself();
+
+		if (!$userData) {
+			// User doesn't exist, but is logged in ❓❓❓
+			return;
+		}
+
+		$userType = intval($userData['type']);
+
+		if($userType != USER_TYPE_SPRT_OFCR){
+			throw new ErrorException("Wrong permissions");
+		}
+
+		$query = 'SELECT L.*, C.name AS courseName, C.code AS courseCode, R.name AS roomName FROM lectures AS L, rooms AS R, (SELECT name,code FROM courses WHERE ID = :courseId) AS C WHERE L.room_id = R.ID AND L.course_id = :courseId';
+
+		// Filter out cancelled lectures
+		$query .= ' AND settings & :cancelled = 0';
+
+		// Add optional ranges to query
+		if (null !== $startTs) {
+			$query .= ' AND start_ts >= :startDate';
+		}
+		if (null !== $endTs) {
+			$query .= ' AND start_ts <= :endDate';
+		}
+
+		// Get list of lectures, ordered
+		$query .= ' ORDER BY start_ts, ID ASC';
+		$stmt = $pdo->prepare($query);
+		$stmt->bindValue(':courseId', $courseId, PDO::PARAM_INT);
```

Comment:
```suggestion
		$stmt->bindValue(':courseId', intval($courseId), PDO::PARAM_INT);
```

Reply:
the intval is useless, since is specified the PARAM_INT. PDO will apply intval by itself